oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2TokenEndpointFilter.java

```diff
@@ -204,9 +204,8 @@ public Authentication convert(HttpServletRequest request) {
 				if (parameters.get(OAuth2ParameterNames.CLIENT_ID).size() != 1) {
 					throwError(OAuth2ErrorCodes.INVALID_REQUEST, OAuth2ParameterNames.CLIENT_ID);
 				}
-			} else {
-				clientPrincipal = SecurityContextHolder.getContext().getAuthentication();
 			}
+			clientPrincipal = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
Moving this line of code does not change any behaviour from the original. As an FYI, if you feel this is a bug, a test must be provided to demonstrate the bug.